core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerRequestCompletionHandler.scala

```diff
@@ -34,16 +34,39 @@ class TransactionMarkerRequestCompletionHandler(brokerId: Int,
   override def onComplete(response: ClientResponse): Unit = {
     val correlationId = response.requestHeader.correlationId
     if (response.wasDisconnected) {
-      trace(s"Cancelled request $response due to node ${response.destination} being disconnected")
-      // re-enqueue the markers
+      trace(s"Cancelled write marker request $response due to node ${response.destination} being disconnected")
       for (txnIdAndMarker: TxnIdAndMarkerEntry <- txnIdAndMarkerEntries) {
+        val transactionalId = txnIdAndMarker.txnId
         val txnMarker = txnIdAndMarker.txnMarkerEntry
-        txnMarkerChannelManager.addTxnMarkersToBrokerQueue(txnIdAndMarker.txnId,
-          txnMarker.producerId(),
-          txnMarker.producerEpoch(),
-          txnMarker.transactionResult(),
-          txnMarker.coordinatorEpoch(),
-          txnMarker.partitions().toSet)
+
+        txnStateManager.getTransactionState(transactionalId) match {
+          case None =>
+            info(s"Transaction topic partition for $transactionalId may likely has emigrated, as the corresponding metadata do not exist in the cache" +
```

Comment:
Mentioned this in a previous patch, but "may likely has" should be "has likely." Note also that we don't need to speculate whether this is the actually the case since we can check whether the partition is still assigned.